core/src/services/web-components.js

```diff
@@ -324,6 +327,26 @@ class WebComponentSvcClass {
       });
     });
   }
+
+  /**
+   * Get stored user settings for a specific user settings group
+   * @param {Object} wc node object definition
+   * @returns a promise that gets resolved with the stored user settings for a specific user settings group.
+   */
+  getUserSettingsForWc(wc) {
+    return new Promise(resolve => {
+      if (wc.userSettingsGroup) {
+        const userSettingsGroupName = wc.userSettingsGroup;
+        LuigiConfig.readUserSettings().then(storedUserSettingsData => {
+          const hasUserSettings =
+            userSettingsGroupName && typeof storedUserSettingsData === 'object' && storedUserSettingsData !== null;
+
+          let userSettings = hasUserSettings ? storedUserSettingsData[userSettingsGroupName] : null;
+          resolve(userSettings);
+        });
+      }

```

Comment:
**if (wc.userSettingsGroup)**  statement above seems to only handle the promise if userSettingsGroup is defined. Should we also maybe handle the other scenario. Maybe this makes it easier for devs to debug their issues in the future. Thought of **reject('userSettingsGroup is not defined');** but maybe also empty rejection would be enough
```suggestion
      } else {
          reject();
      }
```